### PR TITLE
Adjust permissions check on creating PR to fix #6302

### LINKED ISSUE #6302: compare/create PR produces a 404
- Gitea version (or commit ref): 1.7.3
- Git version: 2.21.0
- Operating system: CentOS 7
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [X] No
  - [ ] Not relevant
- Log gist:

## Description
when trying to create a PR gitea shows a 404.
Happens both on
- clicking on the "compare" button
- using the api https://myserver.example/api/v1/repos/myowner/myrepo/pulls

No relevant entries in the log files.

After downgrading to 1.7.2 everything works fine again.

### PR DESCRIPTION
**Pull request by zeripath**
Some users in #6302 appear to be having difficulty creating pull requests since #6098. 

I'm uncertain as to why this additional permissions check is required perhaps these users have a different permissions model but I'm raising this PR as possible fix.

Fixes #6302

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

File: routers/api/v1/repo/pull.go [hunk 1]
```diff
@@ -671,8 +671,8 @@ func parseCompareInfo(ctx *context.APIContext, form api.CreatePullRequestOption)
 		ctx.ServerError("GetUserRepoPermission", err)
 		return nil, nil, nil, nil, "", ""
 	}
-	if !perm.CanReadIssuesOrPulls(true) {
-		log.Trace("ParseCompareInfo[%d]: cannot create/read pull requests", baseRepo.ID)
+	if !perm.CanReadIssuesOrPulls(true) && !perm.CanWrite(models.UnitTypeCode) {
```
Comment (lunny) [inline routers/api/v1/repo/pull.go:674]: It seems it's not reasonable.

**Review by lunny — CHANGES_REQUESTED** [review]

File: routers/api/v1/repo/pull.go (hunk 1 above)
Comment (zeripath) [reply routers/api/v1/repo/pull.go:674]: I don't understand. Are you saying that this is wrong?

What should the permissions test be?

Comment (lunny) [reply routers/api/v1/repo/pull.go:674]: Just a note: For a `parseCompare` we should have two permissions, one is for base repo, another is to head repo.

Comment (zeripath) [reply routers/api/v1/repo/pull.go:674]: OK So I'm not sure that's what's currently being tested.

**Comment by zeripath** [comment]
This is unnecessary the likely issue is something to do with a migration.

**Pull request closed without merging**